src/Reflection/ReflectionParameter.php

```diff
@@ -88,89 +82,46 @@ public function __call($method, $args)
         throw new Exception\BadMethodCallException('Invalid reflection method');
     }
 
-    /**
-     * Retrieve parameter type
-     *
-     * @return string
-     */
-    public function getType()
+    public function getType(): string
     {
         return $this->type;
     }
 
-    /**
-     * Set parameter type
-     *
-     * @param string|null $type
-     * @throws Exception\InvalidArgumentException
-     * @return void
-     */
-    public function setType($type)
+    public function setType(?string $type = null): void
     {
-        if (! is_string($type) && (null !== $type)) {
-            throw new Exception\InvalidArgumentException('Invalid parameter type');
-        }
-
         $this->type = $type;
     }
 
-    /**
-     * Retrieve parameter description
-     *
-     * @return string
-     */
-    public function getDescription()
+    public function getDescription(): ?string
     {
         return $this->description;
     }
 
-    /**
-     * Set parameter description
-     *
-     * @param string|null $description
-     * @throws Exception\InvalidArgumentException
-     * @return void
-     */
-    public function setDescription($description)
+    public function setDescription(?string $description = null): void
     {
-        if (! is_string($description) && (null !== $description)) {
-            throw new Exception\InvalidArgumentException('Invalid parameter description');
-        }
-
         $this->description = $description;
     }
 
-    /**
-     * Set parameter position
-     *
-     * @param int $index
-     * @return void
-     */
-    public function setPosition($index)
+    public function setPosition(?int $index = null): void
```

Comment:
also, the same here: do we want nullable types? please remove default param type on setter.

Reply:
Done, accepting only `int` values now.